shark-android/src/main/java/shark/AndroidObjectInspectors.kt

```diff
@@ -594,4 +612,21 @@ internal fun HeapInstance.unwrapActivityContext(): HeapInstance? {
     }
   }
   return null
+}
+
+/**
+ * Same as [HeapInstance.readField] but throws if the field doesnt exist
+ */
+internal fun HeapInstance.getOrThrow(
+  declaringClassName: String,
+  fieldName: String
+): HeapField {
+  return this[declaringClassName, fieldName] ?: throw IllegalStateException(
+      """
+The heap dump may contain obfuscated class names. If that's the case then the heap analysis 
+is unable to proceed without a mapping file to deobfuscate class names. You can turn off
+obfuscation or run LeakCanary on obfuscated builds by following the instructions at 
+https://square.github.io/leakcanary/recipes/#using-leakcanary-with-obfuscated-apps
+      """
```

Comment:
Let's add a bit more context, e.g.:

```
"""
$instanceClassName is expected to have a $declaringClassName.$fieldName field which cannot be found. This might be due to the app code being obfuscated. If that's the case, then the heap analysis 
is unable to proceed without a mapping file to deobfuscate class names. You can run LeakCanary on obfuscated builds by following the instructions at https://square.github.io/leakcanary/recipes/#using-leakcanary-with-obfuscated-apps
"""
```